app/controllers/concerns/application_multifactor_methods.rb

```diff
@@ -4,8 +4,14 @@ module ApplicationMultifactorMethods
   included do
     def redirect_to_new_mfa
       message = t("multifactor_auths.setup_required_html")
+
+      if request.path_info == edit_settings_path
```

Comment:
Maybe [current_page?](https://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html#method-i-current_page-3F) could be used in here.

Reply:
Wasn't able to get current_page to work here, I believe it's because this is a concern and doesn't have direct access to ActiovView perhaps 🤔

Reply:
Yeah, you would need to include the `ActionView::Helpers::UrlHelper` module into the concern.

```
module ApplicationMultifactorMethods
  include ActionView::Helpers::UrlHelper
```

Reply:
Gave this another attempt. Unfortunately given that this concern is used in many places, including `ActionView::Helpers::UrlHelper` causes some bad behaviour in other controllers relying on it. So given that the current `request.path_info == edit_settings_path` does do what we need it do, will keep it as is.